### PR TITLE
feat(require-2fa): Require 2FA setup before joining organization

### PR DESCRIPTION
**Pull request by maheskett**
screen shot 2018-07-31 at 3 43 18 pm

screen shot 2018-07-31 at 3 44 57 pm

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const Icon = styled(InlineSvg)`
 
 class BackButton extends React.Component {
   static propTypes = {
-    organization: SentryTypes.Organization,
+    organization: PropTypes.object,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: Does this need to be changed?

Comment (maheskett) [reply src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: The organization.id field is required, which creates a false warning now that users can get to account settings without an organization. What do you think about removing `isRequired` here: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/sentryTypes.jsx#L308

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: Wouldn't organization be undefined in that case?

Comment (maheskett) [reply src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: Hm... ya I think you're right. Not sure why I was seeing that warning before. Will change it back.

File: src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx [hunk 2]
```diff
@@ -53,6 +54,20 @@ class BackButton extends React.Component {
         ? t('Project')
         : t('Organization');
 
+    // if the user needs to setup 2fa as part of the org invite flow,
+    // send them back to accept the invite
+    let pendingInvite = Cookies.get('pending-invite');
+    let shouldGoBackToInvite = pendingInvite && !lastAppContext;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: I feel like "pending invite" should disregard `lastAppContext`, I suppose this isn't necessarily the case when you are already a member of other organizations... thoughts?

**Comment by billyvg** [comment]
Frontend looks good to me

File: src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx (hunk 2 above)
Comment (maheskett) [reply src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: I was mildly worried about cookies not getting deleted so added that check... not opposed to getting rid of it though. Would lastAppContext always be falsey coming from the invite flow, whether or not you're a member of another organization?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: Yeah if they come directly from the invite flow, it'll always be null

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**